### PR TITLE
CAMEL-10169: support service lookup by service.pid in OSGi environment

### PR DESCRIPTION
**Pull request by borcsokj**
Created patch to support service lookup by service.pid in OSGi environment

### CODE REVIEW

File: components/camel-core-osgi/src/main/java/org/apache/camel/core/osgi/OsgiServiceRegistry.java [hunk 1]
```diff
@@ -62,18 +67,29 @@ public OsgiServiceRegistry(BundleContext bc) {
...
+            // trying to lookup service by PID if not found by name
+            try {
+                ServiceReference<?>[] refs = bundleContext.getServiceReferences((String)null, "(" + Constants.SERVICE_PID + "=" + name + ")");
+                if (refs != null && refs.length > 0) {
+                    // just return the first one
+                    sr = refs[0];
+                    serviceReferenceQueue.add(sr);
+                    service = bundleContext.getService(sr);
+                }
+            } catch (InvalidSyntaxException ex) {
```
Comment (davsclaus) [inline components/camel-core-osgi/src/main/java/org/apache/camel/core/osgi/OsgiServiceRegistry.java]: We need to do something bette than log an ERROR. Either its a DEBUG/TRACE logging error to ignore this. Or its something that needs to be wrapped and thrown as runtime exception. And is there any API on OSGi you can use to check if the syntax is correct and then just do a if correct then ... else ignore,

**Comment by borcsokj** [comment]
OSGi filter should be valid in almost all cases, logging DEBUG and ignore lookup by service.pid is better than interrupting request processing otherwise. What do you think?

**Comment by davsclaus** [comment]
Thanks for the PR it has been merged. Do you mind closing this?

**Comment by borcsokj** [comment]
Thank you!

**Pull request closed without merging**